src/output.rs

```diff
@@ -1,33 +1,65 @@
 use std::borrow::Cow;
 use std::io::{self, Write};
 
+use base64::{engine::general_purpose, Engine as _};
+use jiff::Timestamp;
 use lscolors::{Indicator, LsColors, Style};
 
+use crate::cli::OutputFormat;
 use crate::config::Config;
 use crate::dir_entry::DirEntry;
+use crate::exit_codes::ExitCode;
 use crate::fmt::FormatTemplate;
 use crate::hyperlink::PathUrl;
 
 #[cfg(unix)]
 use std::os::unix::fs::PermissionsExt;
 
-enum DetailFormat {
-    Json,
-    Yaml,
-}
-
 fn replace_path_separator(path: &str, new_path_separator: &str) -> String {
     path.replace(std::path::MAIN_SEPARATOR, new_path_separator)
 }
 
+#[cfg(unix)]
+fn encode_path(path: &std::path::Path) -> PathEncoding {
+    use std::os::unix::ffi::OsStrExt;
+    let bytes = path.as_os_str().as_bytes();
```

Comment:
As of rust 1.74 we can actually use [as_encoded_bytes](https://doc.rust-lang.org/stable/std/ffi/struct.OsStr.html#method.as_encoded_bytes) to get a slice of bytes in a platform-independent way.